src/sdl2_compat.c

```diff
@@ -3527,153 +3527,89 @@ GestureProcessEvent(const SDL_Event *event3)
                 break;
-            case (4):
+            case 4:
                 pixel_value = *(Uint32 *)pixel;
                 break;
             }
             SDL3_GetRGBA(pixel_value, shape->format, &r, &g, &b, &alpha);
             switch (mode.mode) {
             case (ShapeModeDefault):
-                mask_value = (alpha >= 1 ? 1 : 0);
+                mask_value = (alpha >= 1 ? 0xFFFFFFFF : 0);
                 break;
             case (ShapeModeBinarizeAlpha):
-                mask_value = (alpha >= mode.parameters.binarizationCutoff ? 1 : 0);
+                mask_value = (alpha >= mode.parameters.binarizationCutoff ? 0xFFFFFFFF : 0);
                 break;
             case (ShapeModeReverseBinarizeAlpha):
-                mask_value = (alpha <= mode.parameters.binarizationCutoff ? 1 : 0);
+                mask_value = (alpha <= mode.parameters.binarizationCutoff ? 0xFFFFFFFF : 0);
                 break;
             case (ShapeModeColorKey):
                 key = mode.parameters.colorKey;
-                mask_value = ((key.r != r || key.g != g || key.b != b) ? 1 : 0);
+                mask_value = ((key.r != r || key.g != g || key.b != b) ? 0xFFFFFFFF : 0);
                 break;
             }
-            bitmap_scanline[x / ppb] |= mask_value << (x % ppb);
+            pixels[x] = mask_value;
         }
-    }
-
-    if (SDL_MUSTLOCK(shape)) {
-        SDL3_UnlockSurface(shape);
+        pixels = (Uint32 *)((Uint8 *)pixels + pitch);
     }
 }
 
 
-
 DECLSPEC SDL_Window * SDLCALL
 SDL_CreateShapedWindow(const char *title, unsigned int x, unsigned int y, unsigned int w, unsigned int h, Uint32 flags)
 {
-    SDL_Window *window;
-    int hidden = flags & SDL_WINDOW_HIDDEN;
-
-    CheckEventFilter();
+    flags |= (SDL_WINDOW_HIDDEN | SDL_WINDOW_TRANSPARENT);
 
-    if (g_shaped_window != NULL) {
-        SDL3_SetError("only 1 shaped window");
-        return NULL;
-    }
-
-    flags &= ~SDL2_WINDOW_SHOWN;
-    flags |= SDL_WINDOW_HIDDEN;
-    flags |= SDL_WINDOW_TRANSPARENT;
-
-    window = SDL3_CreateWindow(title, (int)w, (int)h, flags);
-    if (window) {
-        if (!SDL_WINDOWPOS_ISUNDEFINED(x) || !SDL_WINDOWPOS_ISUNDEFINED(y)) {
-            SDL3_SetWindowPosition(window, (int)x, (int)y);
-        }
-        if (!hidden) {
-            SDL3_ShowWindow(window);
-        }
-    }
-
-    shaped_window_cleanup();
-    g_shaped_window = window;
-
-    return window;
+    return SDL_CreateWindow(title, (int)x, (int)y, (int)w, (int)h, flags);
 }
 
 DECLSPEC SDL_bool SDLCALL
 SDL_IsShapedWindow(const SDL_Window *window)
 {
-    if (window == NULL) {
-        return SDL_FALSE;
-    }
-    if (window == g_shaped_window) {
+    if (SDL3_GetWindowFlags((SDL_Window *)window) & SDL_WINDOW_TRANSPARENT) {
         return SDL_TRUE;
     }
     return SDL_FALSE;
 }
 
+#define PROP_WINDOW_SHAPE_MODE_POINTER "sdl2-compat.window.shape_mode"
+
+static void CleanupFreeableProperty(void *userdata, void *value)
```

Comment:
This one must be `SDLCALL` -- i.e.:
`static void SDLCALL CleanupFreeableProperty(void *userdata, void *value)`

Reply:
Fixed